apps/api/src/tasks/tasks.service.ts

```diff
@@ -72,9 +72,23 @@ export class TasksService {
       this.taskModel
     );
 
+    const tpcIds = tasks.map((t) => t.tpc_ids);
+
+    const callsByTasks = [];
+
+    for (const tpcId of tpcIds) {
+      const calls = (await this.calldriversModel.getCallsByTpcId(dateRange, tpcId)).reduce((a, b) => a + b.calls, 0);
+      callsByTasks.push({calls});
+    }
```

Comment:
Using an object instead of an array is highly preferable because you can use it to look up the values using the task ids rather than relying on array indexes being lined up, which is pretty unreliable.

e.g. something like:
```suggestion
    const callsByTasks: { [string]: number } = {};
    
    for (const task of tasks as Task[]) {
      const calls = (await this.calldriversModel.getCallsByTpcId(dateRange, task.tpc_ids)).reduce((a, b) => a + b.calls, 0);
      
      callsByTasks[task._id.toString()] = calls;
    }
    
    const task = tasks.map((task) => ({
      ...task,
      calls: callsByTasks[task._id.toString()] ?? 0
    }));
```